_inc/lib/core-api/wpcom-endpoints/service-api-keys.php

```diff
@@ -268,6 +284,83 @@ public static function validate_service_api_key_mapbox( $key ) {
 		);
 	}
 
+	/**
+	 * Get the site's own Mapbox API key if set, or the WordPress.com's one otherwise.
+	 *
+	 * @return array An array containing the key (if any) and its source ("site" or "wpcom").
+	 */
+	public static function get_service_api_key_mapbox() {
+		// If the site provides its own Mapbox API key, return it.
+		$service_api_key = Jetpack_Options::get_option( self::key_for_api_service( 'mapbox' ) );
+		if ( $service_api_key ) {
+			return self::format_api_key( $service_api_key );
+		}
+
+		// If the site is not WordPress.com, return an empty API key.
```

Comment:
```suggestion
		// If the site is not WordPress.com, not an Atomic site, or not connected to WordPress.com yet, return an empty API key.
```

Reply:
JP sites, regardless if connected or otherwise, are not supposed to receive the token, so this literally is only for WPCOM sites, and to me "WordPress.com sites" is enough to describe "Simple and Atomic sites".

Please let me know if I'm wrong here, I don't have any strong opinions about it, and if you think that "not WordPress.com and not Atomic" is more clear, I'll be happy to change the comment.

Reply:
> to me "WordPress.com sites" is enough to describe "Simple and Atomic sites".

Yeah, it really boils down to how you describe "WordPress.com site". To me that only includes simple sites. That's often what you'll find in the Jetpack codebase as well, where "WordPress.com site" is most often looked at as `defined( 'IS_WPCOM' ) && IS_WPCOM`.

It's not a big deal here and not a blocker, but I think being more verbose would be better in general.